cpp/src/join/quadtree_point_in_polygon.cu

```diff
@@ -0,0 +1,573 @@
+  auto d_point_index_col = cudf::mutable_column_device_view::create(*point_index_col, stream);
+
+  quad_pip_phase2_kernel<T><<<num_valid_pair, threads_per_block, 0, stream>>>(d_pq_poly_idx.data(),
+                                                                              d_pq_quad_idx.data(),
+                                                                              d_quad_offset.data(),
+                                                                              d_quad_length.data(),
+                                                                              *d_quad_offsets,
+                                                                              *d_point_indices,
+                                                                              *d_point_x,
+                                                                              *d_point_y,
+                                                                              *d_poly_offsets,
+                                                                              *d_ring_offsets,
+                                                                              *d_poly_points_x,
+                                                                              *d_poly_points_y,
+                                                                              d_num_hits.data(),
+                                                                              *d_poly_index_col,
+                                                                              *d_point_index_col);
+
+  CUDA_TRY(cudaStreamSynchronize(stream));
+
+  std::vector<std::unique_ptr<cudf::column>> cols{};
+  cols.reserve(2);
+  cols.push_back(std::move(poly_index_col));
+  cols.push_back(std::move(point_index_col));
+  return std::make_unique<cudf::table>(std::move(cols));
+}
+
+struct dispatch_quadtree_point_in_polygon {
+  template <typename T, std::enable_if_t<std::is_floating_point<T>::value> * = nullptr>
+  std::unique_ptr<cudf::table> operator()(cudf::table_view const &poly_quad_pairs,
+                                          cudf::table_view const &quadtree,
+                                          cudf::column_view const &point_indices,
+                                          cudf::column_view const &point_x,
+                                          cudf::column_view const &point_y,
+                                          cudf::column_view const &poly_offsets,
+                                          cudf::column_view const &ring_offsets,
+                                          cudf::column_view const &poly_points_x,
+                                          cudf::column_view const &poly_points_y,
+                                          rmm::mr::device_memory_resource *mr,
+                                          cudaStream_t stream)
+  {
+    return compute_quadtree_point_in_polygon<T>(poly_quad_pairs.column(0),
```

Comment:
Possibly an unnecessary layer? Could move body of `compute_quadtree_point_in_polygon` here?